IdentityCore/src/webview/response/MSIDWebOAuth2Response.m

```diff
@@ -26,8 +26,127 @@
 //------------------------------------------------------------------------------
 
 #import "MSIDWebOAuth2Response.h"
+#import "MSIDWebAADAuthResponse.h"
+#import "MSIDWebWPJAuthResponse.h"
 
 @implementation MSIDWebOAuth2Response
 
++ (MSIDWebOAuth2Response *)responseWithURL:(NSURL *)url
+                              requestState:(NSString *)requestState
+                             stateVerifier:(MSIDWebUIStateVerifier)stateVerifier
+                                   context:(id<MSIDRequestContext>)context
+                                     error:(NSError **)error;
+
+{
+    // This error case *really* shouldn't occur. If we're seeing it it's almost certainly a developer bug
+    if ([NSString msidIsStringNilOrBlank:url.absoluteString])
+    {
+        if (error){
+            *error = MSIDCreateError(MSIDOAuthErrorDomain, MSIDErrorNoAuthorizationResponse, @"No authorization response received from server.", nil, nil, nil, context.correlationId, nil);
+            return nil;
+        }
+    }
+    
+    // Check for WPJ response
+    if([url.absoluteString hasPrefix:@"msauth://"])
```

Comment:
I think "msauth" is something really AAD specific, so let's discuss if we should keep it here.

Reply:
I'm with you, I'd rather have this in a initializer for MSIDWebAADAuthResponse and check for nil there.
@jasoncoolmax what do you think?